### DISCUSSION TITLE
May I filters records by m2m relation fileld in one query(without "for" cycle) in Golang?

### DISCUSSION THREAD

**Discussion by hel8er** [pocketbase/pocketbase#3293] [Q&A] [answered]
May I filters records by m2m relation fileld in one query(without "for" cycle) in Golang?

**Comment by ganigeorgiev** [comment]
I'm not entirely sure that I understand the question.

If you want to search based on a relation field with a custom query you can perform a `JOIN` with the related collection.

The only caveat is if you have a _multiple_ `relation` field because in this case the related ids are stored as serialized JSON array and you have to perform an extra [`json_each`](https://www.sqlite.org/json1.html#jeach) to match the ids with the joined relation records.

For example, if you have a collection _"foo"_ with a multiple `relation` field _"barRelField"_ pointing to _"bar"_ collection and you want to search by some field in the _"bar"_ collection:

```go
records := []*models.Record{}

err := app.Dao().RecordQuery("foo").
	Distinct(true).
	InnerJoin("json_each(foo.barRelField) je", nil).
	InnerJoin("bar", dbx.NewExp("bar.id = je.value")).
	AndWhere(dbx.HashExp{"bar.something": "active"}).
	Limit(10).
	All(&records)
```

If that is not what you are looking for, please elaborate.

**Reply by ganigeorgiev** [reply]
Depending on your use case, another alternative could be also to use the filter query syntax [`Dao.FindRecordsByFilter`](https://pkg.go.dev/github.com/pocketbase/pocketbase/daos#Dao.FindRecordsByFilter):

```go
records, err := app.Dao().FindRecordsByFilter(
	"foo",
	"barRelField.something ?= {:something}",
	"", // optional sort
	10, // limit
	0,  // offset
	dbx.Params{"something": "active"},
)
```

**Reply by hel8er** [reply]
```
func testDB(userId string) {
	// TgUsersCollection, _ := App.Dao().FindCollectionByNameOrId("tg_users")
	records, _ := App.Dao().FindRecordsByFilter(
		"keys",                     // collection
		"user = {:user}",           // filter
		"-publised",                // sort
		10,                         // limit
		0,                          // offset
		dbx.Params{"user": userId}, // optional filter params
	)
	log.Println(records) # prints "[ ]"

}


```
user field is a slice from users ids, but isnt string of one id

**Reply by ganigeorgiev** [reply, marked as answer]
Use dot-notation in similar manner to the example above, aka. you can try changing the filter expression to:

```
user.id ?= {:user}
```
`?=` is used as operator in order to perform "at-least-one-of/any" match.

**Reply by hel8er** [reply]
it works! Thanx so mush!
